core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -442,6 +442,9 @@ class KafkaApis(val requestChannel: RequestChannel,
     val versionId = request.header.apiVersion
     val clientId = request.header.clientId
 
+    var result = clientId.split('_');
+    var eventPattern = result(1);
```

Comment:
In Kafka 0.11, you can use Record Header to store eventPattern info.

Reply:
Yes, that is my previous plan, but it need additional work to parse it from header. I think this is not the highest priority issue we need to resolve, we could left hack way here and make code work firstly.